datahub-web-react/src/app/entity/user/UserProfile.tsx

```diff
@@ -19,14 +21,69 @@ export default function UserProfile() {
     const { urn, subview, item } = useUserParams();
     const { loading, error, data } = useGetUserQuery({ variables: { urn } });
 
-    if (loading) {
+    const username = data?.corpUser?.username;
+
+    const ownershipResult: any = {};
+
+    ownershipResult[EntityType.Chart] = useGetSearchResultsQuery({
```

Comment:
Since hook cannot be called inside any loop, not sure if there is a way in Typescript that can make it dynamic, i.e. looping through all enum values and get search query

Reply:
We'd like to add support for an `all` endpoint in the future. In the meantime, I'd recommend encapsulating this all in a 2nd hook `useGetEachEntitySearchResults` so that the UserProfile render function isn't so large

Reply:
+1 to wrapping this